lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -499,29 +552,58 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
 
     posix_spawn_file_actions_destroy(&fileActions);
     posix_spawnattr_destroy(&attributes);
-    
+
+    // Close the write end of the release pipe
+    ::close(releasePipe[1]);
+
+
     // If we failed to launch a process, clean up and abort.
     if (pid == -1) {
+      ::close(releasePipe[0]);
+
       if (shouldCaptureOutput) {
         ::close(outputPipe[1]);
         ::close(outputPipe[0]);
       }
-      return wasCancelled ? CommandResult::Cancelled : CommandResult::Failed;
+      auto result = wasCancelled ? CommandResult::Cancelled : CommandResult::Failed;
+      completionFn.unwrapIn([result](CommandCompletionFn fn){fn(result);});
+      return;
     }
 
+    // Set up our select() structures
+    pollfd readfds[] = {
+      { releasePipe[0], POLLIN, NULL },
+      { outputPipe[0], POLLIN, NULL }
```

Comment:
Just for clarity, I would prefer we only set readfds[1].0 if shouldCaptureOuput is true.